unstructured/unstructured.go

```diff
@@ -61,20 +62,17 @@ func extractNestedField(obj interface{}, result map[string]interface{}, field ..
 		return nil
 	}
 	// Intermediate map — recurse
-	childMap := make(map[string]interface{})
+	childMap := map[string]interface{}{}
 	err := extractNestedField(val.Interface(), childMap, field[1:]...)
 	if err != nil {
 		return err
 	}
 	// Merge into parent map
-	if existing, exists := result[field[0]]; exists {
-		if m, ok := existing.(map[string]interface{}); ok {
-			for k, v := range childMap {
-				m[k] = v
-			}
-		}
-	} else {
-		result[field[0]] = childMap
+	if _, ok := result[field[0]]; !ok {
+		result[field[0]] = map[string]interface{}{}
+	}
+	for k, v := range childMap {
+		result[field[0]].(map[string]interface{})[k] = v
```

Comment:
Still wondering if we want an explicit check here to verify that the type is a map. We'll panic here if it's not already, but it would be good to have a more explicit message IMO.